### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/Application.java [hunk 1]
```diff
@@ -1,15 +1,6 @@
 package mate.academy.internetshop;
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/Application.java]: Do we need this class?

File: src/main/java/mate/academy/internetshop/controllers/AddProductToBucketController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.ShoppingCartService;
+
+public class AddProductToBucketController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("mate.academy.internetshop");
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/AddProductToBucketController.java]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internetshop");
```

File: src/main/java/mate/academy/internetshop/controllers/InjectDataController.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.User;
+import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.UserService;
+
+public class InjectDataController extends HttpServlet {
+    private static final Injector INJECTOR = Injector.getInstance("mate.academy.internetshop");
+    private static final UserService USER_SERVICE
+            = (UserService) INJECTOR.getInstance(UserService.class);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/InjectDataController.java]: injector it's constant, but services not

File: src/main/java/mate/academy/internetshop/controllers/ProductListController.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.service.ProductService;
+
+public class ProductListController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/ProductListController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

Comment (dmytro-hryhoruk) [reply src/main/java/mate/academy/internetshop/controllers/ProductListController.java]: let's add verbs to Controller's names for example  GetAllUsersController (here and everywhere)

File: src/main/webapp/WEB-INF/jsp/productCreate.jsp [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+<html>
+<head>
+    <title>Create product</title>
+</head>
+<body>
+<br>
+<a href="${pageContext.request.contextPath}/" style="text-align: left">Main page</a>
+<br>
+<h1>Create new product</h1>
+<h4>${massage}</h4>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/jsp/productCreate.jsp]:
```suggestion
<h4>${message}</h4>
```

File: src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
+        String passwordRepeat = req.getParameter("pwd-repeat");
+        if (password.equals(passwordRepeat)) {
+            USER_SERVICE.create(new User("USER", login, password));
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java]: what is "USER"?

File: src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String price = req.getParameter("price");
+
+        if (name.equals("") || price.equals("") || !price.matches("[0-9]+[.]?[0-9]{0,2}")) {
+            req.setAttribute("massage", "You write wrong data: " + price);
+        } else {
+            PRODUCT_SERVICE.create(new Product(name, Double.parseDouble(price)));
+            req.setAttribute("massage", "Product added : name = " + name + " prise = " + price);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java]: let's remove this checking for now we will handle this situation later in better way

File: src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+            resp.sendRedirect(req.getContextPath() + "/");
+        } else {
+            req.setAttribute("massage", "Your password and repeated password aren`t the same!");
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java]:
```suggestion
            req.setAttribute("message", "Your password and repeated password aren`t the same!");
```

File: src/main/java/mate/academy/internetshop/controllers/UserListController.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.User;
+import mate.academy.internetshop.service.UserService;
+
+public class UserListController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/UserListController.java]: let's add verbs to Controller's names for example  GetAllUsersController (here and everywhere)

File: src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -71,4 +71,13 @@ public ShoppingCart update(ShoppingCart element) {
     public boolean delete(Long id) {
         return cartDao.delete(id);
     }
+
+    @Override
+    public double getTotalPrice(Long id) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java]: whit this method you'll have to ask DB one more time even you would have this shoppingCart by that time

Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java]: changed. now this mrthod dont't call to DB

File: src/main/java/mate/academy/internetshop/controllers/UserListController.java (hunk 9 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/UserListController.java]: done

File: src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java (hunk 7 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java]: done

File: src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java (hunk 6 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java]: added new input for user_name

File: src/main/java/mate/academy/internetshop/controllers/ProductListController.java (hunk 4 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/ProductListController.java]: done

File: src/main/java/mate/academy/internetshop/Application.java (hunk 1 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/Application.java]: removed

File: src/main/java/mate/academy/internetshop/controllers/AddProductToBucketController.java (hunk 2 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/AddProductToBucketController.java]: changed everywhere

File: src/main/webapp/WEB-INF/jsp/productCreate.jsp (hunk 5 above)
Comment (VitaliiLeshchenko) [reply src/main/webapp/WEB-INF/jsp/productCreate.jsp]: renamed=)

File: src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java (hunk 8 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java]: done

File: src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java (hunk 6 above)
Comment (dmytro-hryhoruk) [reply src/main/java/mate/academy/internetshop/controllers/UserRegistrationController.java]: so let client pass this data then

File: src/main/java/mate/academy/internetshop/controllers/ProductAddToBucketController.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.ShoppingCartService;
+
+public class ProductAddToBucketController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/ProductAddToBucketController.java]:
```suggestion
public class AddProductToBucketController extends HttpServlet {
```

File: src/main/java/mate/academy/internetshop/controllers/ProductDeleteFromBucketController.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.ShoppingCartService;
+
+public class ProductDeleteFromBucketController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/ProductDeleteFromBucketController.java]:
```suggestion
public class DeleteProductFromBucketController extends HttpServlet {
```

File: src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.internetshop.lib.Injector;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.service.ProductService;
+
+public class ProductRegistrationController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java]:
```suggestion
public class AddProductController extends HttpServlet {
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Внеси всі зміни, змерджи і приступай до наступного)

File: src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java (hunk 13 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/ProductRegistrationController.java]: done

File: src/main/java/mate/academy/internetshop/controllers/ProductDeleteFromBucketController.java (hunk 12 above)
Comment (VitaliiLeshchenko) [reply src/main/java/mate/academy/internetshop/controllers/ProductDeleteFromBucketController.java]: done

**Pull request merged**